validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -507,6 +507,9 @@ def setup(self, socket_type, complete_or_error_queue):
             self._socket.set(zmq.TCP_KEEPALIVE_IDLE, self._connection_timeout)
             self._socket.set(zmq.TCP_KEEPALIVE_INTVL, self._heartbeat_interval)
 
+            # Enable ZMQ support for IPv6
+            self._socket.setsockopt(zmq.IPV6, 1)
```

Comment:
This might be causing some issues with the docker files specifying `tcp://eth0:4004` etc.  I would suggest putting this behind a flag/configuration option, or detecting the address type before enabling this flag.

Reply:
I'm checking into this now. Thanks.

Reply:
I went ahead and made a modification that resolves this. I now parse the address to see if its a valid IPv6 address, and only set the flag if that is true.

All tests pass now.

Thanks again.